index.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+entries:
+  sail-operator:
+  - apiVersion: v2
+    appVersion: 0.1.0
+    created: "2024-09-23T19:48:26.553639312+03:00"
+    description: A Helm chart for Kubernetes
+    digest: ba77933659946e929da51baecd453a42b731a8a993563543efff275958ca2b98
+    name: sail-operator
+    type: application
+    urls:
+    - https://istio-ecosystem.github.io/sail-operator/sail-operator-0.1.0.tgz
```

Comment:
maybe it's a stupid question, but would it be possible to just link to the release (https://github.com/istio-ecosystem/sail-operator/releases/download/0.1.0/sail-operator-0.1.0.tgz)? then we wouldn't have to pollute our repo with binaries

Reply:
@dgn 

Wow, that's some really interesting approach.

I searched over the official documentation for separate store of `index.yaml` file, which holds the helm artifact metadata and the actual helm artifact, but haven't find a thing.

So, I did some simulation and it actually works.
If I'm placing the path of the release asset artifact under the `url` argument of `index.yaml` file, it's able to pull and install or upgrade helm deployment by using the artifacts from the gh release assets.

But there is one single thing to notice.
I found the following question and answer regarding the helm `urls` argument in the index.yaml file.
https://github.com/helm/helm/issues/7314#issuecomment-569790204

So, turns out that `urls` argument was designed to support a mirroring approach.
But later it was never requested an improvement or any change.
So, they left it in the API for backwards compatibility purposes.

Now, my concern comes from that it could be removed (maybe?) at some point.
So, I'm not sure it's good to implement this solution based on it.
From the other side, this thread is from 2019 and nothing changed since then.

What do you think?

Thanks.

Reply:
it looks to me as though that is more about having >1 url. The downloader will always follow the link. Seems safe enough to me :) I'd say let's go ahead and add the link to the GitHub releases, keeps our git repo small. Thank you for investigating!

Reply:
As @dgn mentioned, the helm charts downloader seems to support [multiple URL formats](https://github.com/helm/helm/tree/main/pkg/downloader/testdata/repository) and for fully-qualified URLs, it simply follows the url.
https://github.com/helm/helm/blob/main/pkg/downloader/chart_downloader_test.go

Reply:
@dgn @sridhargaddam 

Done.